### ISSUE TITLE
Convox does not clean up Release correctly

### ISSUE THREAD

**Issue by nguyendangminh** [convox/kernel#99]
Hi,

It seems that Convox doesn't clean up Releases correctly. After deleting an application then re-creating a new one with the same name, "convox releases" still lists the old releases.

minhnd@swallow:~/workspace/go/convox/sinatra$ convox apps delete sinatra
Deleting sinatra: OK
minhnd@swallow:~/workspace/go/convox/sinatra$ convox apps
APP  STATUS
minhnd@swallow:~/workspace/go/convox/sinatra$ convox releases sinatra
ERROR: no such app: sinatra
minhnd@swallow:~/workspace/go/convox/sinatra$ convox apps create sinatra
Creating app sinatra: OK
minhnd@swallow:~/workspace/go/convox/sinatra$ convox releases sinatra
ID            CREATED       ACTIVE      
RGKTIFXYWFX   37 minutes ago              
RUGXXQQKAJR   1 day ago

**Comment by rjocoleman** [comment]
I think you're encountering #98

**Comment by nguyendangminh** [comment]
Two issues are not the same. As you can see in the logs, I deleted an application 1 day ago but its release is still existing.

**Comment by rjocoleman** [comment]
I don't see anything actually deleting releases (just cleaning up the env per release): https://github.com/convox/kernel/blob/1bb05d324e767e0cb65215507d9dfb0a1d356cc9/models/app.go#L133-L162

There are no calls to https://docs.aws.amazon.com/sdk-for-go/api/service/dynamodb/DynamoDB.html#DeleteItem-instance_method or https://docs.aws.amazon.com/sdk-for-go/api/service/dynamodb/DynamoDB.html#BatchWriteItem-instance_method

**Comment by nguyendangminh** [comment]
Yes, it just deletes files stored in S3 bucket. Convox team may forgeted to delete items in DynamoDB.

**Comment by ddollar** [comment]
Yes, this is an oversight. We will need to create a CustomResource in the convox/app CloudFormation script that deletes the items from DynamoDB during app deletion similar to the S3BucketCleanup

https://github.com/convox/app/blob/master/template/staging.tmpl#L349-L356

**Comment by nguyendangminh** [comment]
How about putting the following into the func Cleanup() in release.go:
```
	// delete release on dynamodb
	req := &dynamodb.DeleteItemInput{
		Key: &map[string]*dynamodb.AttributeValue{
			"id": &dynamodb.AttributeValue{S: aws.String(r.Id)},
		},
		TableName: aws.String(releasesTable(r.App)),	
	}

	_, err = DynamoDB().DeleteItem(req)
	
	if err != nil {
		return err
	}
```

**Comment by ddollar** [comment]
That would work but would not clean up the releases were one to just go in and delete the CloudFormation stack. I'd prefer to keep the stacks as self-contained as possible.

**Comment by nguyendangminh** [comment]
Okay, I'll try to submit a PR that way.
